app/controllers/settings_controller.rb

```diff
@@ -3,9 +3,52 @@
 class SettingsController < ApplicationController
   include Europeana::Styleguide
 
+  ##
+  # @todo move this into two separate actions, one for GET, one for PUT, with
+  #  a resourceful controller and routes, per http://guides.rubyonrails.org/routing.html#singular-resources
   def language
-    respond_to do |format|
-      format.html
+    if request.put?
+      locale_changed = set_locale_from_param
+
+      respond_to do |format|
+        format.html do
+          render action: :language, status: status_from_flash
+        end
+        format.json do
+          render json: { success: !flash.key?(:alert), refresh: locale_changed, message: flash.key?(:alert) ? flash.now[:alert] : flash.now[:notice] },
+            status: status_from_flash
+        end
+      end
+    else
+      respond_to do |format|
+        format.html
+      end
+    end
+  end
+
+  protected
+
+  def status_from_flash
+    flash.key?(:alert) ? 400 : 200
+  end
+
+  def set_locale_from_param
```

Comment:
Assignment Branch Condition size for set_locale_from_param is too high. [24.52/15]<br>Method has too many lines. [14/10]